das/state.go

```diff
@@ -28,6 +28,9 @@ type RoutineState struct {
 	// tracks the square width of the latest successfully sampled
 	// height of the routine
 	LatestSampledSquareWidth uint64 `json:"latest_sampled_square_width"`
+	// amount of sampled headers
+	SampledHeadersAmount uint64 `json:"sampled_header_amount"`
```

Comment:
we don't need to track this on daser IMO unless it's for metrics purposes, but i'm sure tracing will provide a better mechanism for this.